### PR TITLE
All tests working! Many bugs fixed!

### PR DESCRIPTION
**Pull request by tibetsprague**
All Hylo Node tests now pass. Fixes https://github.com/Hylozoic/hylo-node/issues/517

Also a number of bugs were fixed:
- Various issues with unread counts for chat threads and new post counts for group memberships. Fixes https://github.com/Hylozoic/hylo-evo/issues/937
- Duplicate threads getting created when messaging the same group of people
- Correctly export twitter name

### CODE REVIEW

File: api/controllers/CommentController.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ module.exports = {
     )
   },
   createBatchFromEmailForm: function (req, res) {
+    // TODO: fix
```
Comment (thomasgwatson) [inline api/controllers/CommentController.js:41]: Fix for now or later?

File: api/models/User.js [hunk 2]
```diff
@@ -97,12 +97,11 @@ module.exports = bookshelf.Model.extend(merge({
...
 
   posts: function () {
-    return this.belongsToMany(Post).through(PostUser).query(q => q.where(function () {
-      this.where('type', null).orWhere('type', '!=', Post.Type.THREAD)
-    }))
+    return this.hasMany(Post).query(q => q.where('type', '!=', Post.Type.THREAD))
   },
 
   projects: function () {
+    // TODO: fix
```
Comment (thomasgwatson) [inline api/models/User.js:104]: Now or later?

**Review by thomasgwatson — APPROVED** [review]
Checked it over as best I could. Just checking in on a few of the TODOs as to whether that was TODO now or later

File: api/controllers/CommentController.js (hunk 1 above)
Comment (tibetsprague) [reply api/controllers/CommentController.js:42]: oh yeah this is an eventually someday if we ever bring back this feature... not really important enough to track with a ticket yet just wanted to flag in the code that this functionality is not totally working

File: api/models/User.js (hunk 2 above)
Comment (tibetsprague) [reply api/models/User.js:104]: later, whenever we get back to the projects functionality in a real way

**Pull request merged**